proto/v3/message_contents/association.proto

```diff
@@ -6,18 +6,35 @@ package xmtp.v3.message_contents;
 option go_package = "github.com/xmtp/proto/v3/go/v3/message_contents";
 option java_package = "org.xmtp.proto.v3.message.contents";
 
-// Allows for us to update the format of the association text without
-// incrementing the entire proto
-enum AssociationTextVersion {
-    ASSOCIATION_TEXT_VERSION_UNSPECIFIED = 0;
-    ASSOCIATION_TEXT_VERSION_1 = 1;
-}
-
 // EIP191Association is used for all EIP 191 compliant wallet signatures
 message Eip191Association {
-    AssociationTextVersion association_text_version = 1;
+    reserved 1;
     RecoverableEcdsaSignature signature = 2;
-    string wallet_address = 3;
+    oneof association_data {
+        CreateIdentityAssociationData create_identity_data = 3;
+        InstallationGrantAssociationData installation_grant_data = 4;
+    }
+
+    // Legacy - all of the information needed to construct the Create Identity
+    // association text used in XMTP v2
+    message CreateIdentityAssociationData {
+        string wallet_address = 1;
```

Comment:
This is confusing to me, since the Create Identity association text is actually the encoded protobuf of the public key, not the wallet address.